master/include/embedded/digitalReceiver.hpp

```diff
@@ -4,126 +4,133 @@
+
+    void askReadWatchdog() const;
+
+    void updateLeftWheelRpm() const;
 };
 
-void DigitalReceiver::updateLeftWheelRpm() {
-  digitalData->_left_wheel_rpm =
-      digitalData->pulse_count /
-      (WHEEL_MEASUREMENT_INTERVAL_MIN * PULSES_PER_ROTATION);
+inline void DigitalReceiver::updateLeftWheelRpm() const {
+    digitalData->_left_wheel_rpm =
+            digitalData->pulse_count /
+            (WHEEL_MEASUREMENT_INTERVAL_MIN * PULSES_PER_ROTATION);
 
-  digitalData->pulse_count = 0;
-  digitalData->left_wheel_update_ts.update();
+    digitalData->pulse_count = 0;
+    digitalData->left_wheel_update_ts.check();
 }
 
-Button DigitalReceiver::newButton(uint8_t pin) {
-  Button button = Button();
-  button.attach(pin, INPUT_PULLUP);
-  button.interval(DEBOUNCE_INTERVAL);
-  button.setPressedState(PRESSED_STATE);
+inline Button DigitalReceiver::newButton(uint8_t pin) {
+    Button button;
+    button.attach(pin, INPUT_PULLUP);
+    button.interval(DEBOUNCE_INTERVAL);
+    button.setPressedState(PRESSED_STATE);
 
-  return button;
+    return button;
 }
 
-void DigitalReceiver::digitalReads() {
-  readLwss();
-  readPneumaticLine();
-  readMission();
-  readAsmsSwitch();
-  readAatsSwitch();
-  askReadWatchdog();
+inline void DigitalReceiver::digitalReads() {
+    readLwss();
+    readPneumaticLine();
+    readMission();
+    readAsmsSwitch();
+    readAatsSwitch();
+    askReadWatchdog();
 }
 
-void DigitalReceiver::readLwss() {
-  bool const current_lwss_state = digitalRead(LWSS_PIN);
+inline void DigitalReceiver::readLwss() const {
+    bool const current_lwss_state = digitalRead(LWSS_PIN);
 
-  if (current_lwss_state == HIGH && digitalData->last_lwss_state == LOW)
-    digitalData->pulse_count++;
+    if (current_lwss_state == HIGH && digitalData->last_lwss_state == LOW)
+        digitalData->pulse_count++;
 
-  digitalData->last_lwss_state = current_lwss_state;
+    digitalData->last_lwss_state = current_lwss_state;
 
-    if (digitalData->left_wheel_update_ts.hasTimedOut(
-        WHEEL_MEASUREMENT_INTERVAL_MIN))  // TODO André: I believe this won't work because it's in minutes, not ms
+    if (digitalData->left_wheel_update_ts.check())
         updateLeftWheelRpm();
 }
 
-void DigitalReceiver::readPneumaticLine() {
-  // TODO: wait for eletro indications
+inline void DigitalReceiver::readPneumaticLine() const {
+    // TODO: wait for eletro indications
 }
 
-void DigitalReceiver::readMission() {
-  // Enum value attributed considering the True Boolean Value
-  *mission = static_cast<Mission>(
-      digitalRead(MISSION_MANUAL_PIN) * MANUAL |
-      digitalRead(MISSION_ACCELERATION_PIN) * ACCELERATION |
-      digitalRead(MISSION_SKIDPAD_PIN) * SKIDPAD |
-      digitalRead(MISSION_AUTOCROSS_PIN) * AUTOCROSS |
-      digitalRead(MISSION_TRACKDRIVE_PIN) * TRACKDRIVE |
-      digitalRead(MISSION_EBSTEST_PIN) * EBS_TEST |
-      digitalRead(MISSION_INSPECTION_PIN) * INSPECTION);
+inline void DigitalReceiver::readMission() const {
```

Comment:
why the consts in these functions? Using it in a read/set function may give the false idea that this is not changing any variable

Reply:
clion suggestions, they can be const as they are not editing any class attributes but rather (the pointer adress i.e), but we can remove this